opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -255,11 +259,78 @@ public Boolean checkIfExists(Table table, String baseEntityId, SQLiteDatabase sq
         return false;
     }
 
+    private List<String> getFormSubmissionIdsFromJsonArray(JSONArray array) {
+        if (array != null && array.length() != 0) {
+            List<String> stringList = new ArrayList<>();
+            for (int i = 0; i < array.length(); i++) {
+                JSONObject jsonObject = array.optJSONObject(i);
+                if (jsonObject != null) {
+                    String formSubmissionId = jsonObject.optString(event_column.formSubmissionId.name());
+                    if (StringUtils.isNotBlank(formSubmissionId)) {
+                        stringList.add(formSubmissionId);
+                    }
+                }
+            }
+            return stringList;
+        }
+        return new ArrayList<>();
+    }
+
+    protected void populateFormSubmissionIds(@NonNull List<String> formSubmissionIdsList,
+                                             Set<String> formSubmissionIds) {
+        populateFormSubmissionIds(formSubmissionIdsList, formSubmissionIds, null);
+    }
+
+    protected void populateFormSubmissionIds(@NonNull List<String> formSubmissionIdsList,
+                                             Set<String> formSubmissionIds, @Nullable Integer pageSize) {
+        Integer tempPageSize = pageSize;
+
+        if (tempPageSize == null) {
+            tempPageSize = FORM_SUBMISSION_IDS_PAGE_SIZE;
+        }
+
+        String query = "SELECT "
+                + EventClientRepository.event_column.formSubmissionId
+                + " FROM event";
+
+        List<String> tempList;
+
+        boolean shouldEnd = false;
+
+        if (formSubmissionIdsList.size() <= tempPageSize) {
+            tempList = formSubmissionIdsList;
+            shouldEnd = true;
+        } else {
+            tempList = formSubmissionIdsList.subList(0, tempPageSize);
+        }
+
+        int limit = formSubmissionIdsList.size() - tempPageSize;
+
+        query += " WHERE " + EventClientRepository.event_column.formSubmissionId + " IN ( " + StringUtils.repeat("?", ", ", tempList.size()) + ")";
+
+        try (Cursor mCursor = getReadableDatabase().rawQuery(query, tempList.toArray(new String[0]))) {
+            if (mCursor != null) {
+                while (mCursor.moveToNext()) {
+                    formSubmissionIds.add(mCursor.getString(0));
+                }
+            }
+        } catch (SQLException e) {
+            Timber.e(e);
+        }
+
+        if (shouldEnd) {
+            return;
+        }
+
+        populateFormSubmissionIds(formSubmissionIdsList.subList(tempPageSize, tempPageSize + limit), formSubmissionIds, tempPageSize);
```

Comment:
is this correct `formSubmissionIdsList.subList(tempPageSize, tempPageSize + limit)`